File: system-http-client/src/main/java/com/levnovikov/core_network/CallExecutor.java
package com.levnovikov.core_network;

import com.levnovikov.core_network.request.Request;
import com.levnovikov.core_network.response.Response;

import java.io.IOException;

/**
 * Created by lev.novikov
 * Date: 27/3/18.
 */

public interface CallExecutor {
    Response execute(Request request) throws IOException;
}

File: system-http-client/src/main/java/com/levnovikov/core_network/CallExecutor.kt
package com.levnovikov.core_network

import com.levnovikov.core_network.request.Request
import com.levnovikov.core_network.response.Response

import java.io.IOException

/**
 * Created by lev.novikov
 * Date: 27/3/18.
 */

interface CallExecutor {
    @Throws(IOException::class)
    fun execute(request: Request): Response
}
